pkg/instancetype/revision/store_test.go

```diff
@@ -201,7 +228,19 @@ var _ = Describe("Instancetype and Preferences revision handler", func() {
 				Expect(err).ToNot(HaveOccurred())
 
 				Expect(storeHandler.Store(vm)).To(Succeed())
-				Expect(vm.Spec.Instancetype.RevisionName).To(Equal(instancetypeControllerRevision.Name))
+
+				Expect(vm.Spec.Instancetype.RevisionName).To(BeEmpty())
+				Expect(vm.Status.InstancetypeRef.Name).To(Equal(clusterInstancetype.Name))
+				Expect(vm.Status.InstancetypeRef.Kind).To(Equal(clusterInstancetype.Kind))
+				Expect(vm.Status.InstancetypeRef.ControllerRevisionRef.Name).To(Equal(instancetypeControllerRevision.Name))
+
+				updatedVM, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(updatedVM.Spec.Instancetype.RevisionName).To(BeEmpty())
+				Expect(updatedVM.Status.InstancetypeRef.Name).To(Equal(clusterInstancetype.Name))
+				Expect(updatedVM.Status.InstancetypeRef.Kind).To(Equal(clusterInstancetype.Kind))
+				Expect(updatedVM.Status.InstancetypeRef.ControllerRevisionRef.Name).To(Equal(instancetypeControllerRevision.Name))
```

Comment:
Why get VM again?

Reply:
The initial assertions are just asserting the state of the `vm` passed into `Store`. These repeat assertions are against the object as held by the fakeclient. I can also assert that `PatchStatus` has been called here I guess but asserting the state of the object in the fakeclient might also be enough.